apps/api/src/donations/donations.service.ts

```diff
@@ -114,6 +115,87 @@ export class DonationsService {
     return donation
   }
 
+  /**
+   * Create initial donation object for tracking purposes
+   * This is used when the payment is created from the payment intent
+   */
+  async createInitialDonationFromIntent(
+    campaign: Campaign,
+    stripePaymentDto: CreateStripePaymentDto,
+    paymentIntent: Stripe.PaymentIntent,
+  ): Promise<Donation> {
```

Comment:
This is basically a copy of `createInitialDonationFromSession` adapted to work with the `stripePaymentDto` and the new requests.